File: shoring/compat/src/main/java/com/panopset/compat/JavaVersionChart.java
package com.panopset.compat;

import java.io.StringWriter;

public class JavaVersionChart {

  public static void main(String... args) {
    Logop.dspmsg(getChart());
  }

  public static String getChart() {
    return new JavaVersionChart().print();
  }

  private String print() {
    StringWriter sw = new StringWriter();
    for (MajorVersion mv : MajorVersion.values()) {
      sw.append(Stringop.EOL);
      sw.append(mv.toString());
    }
    return sw.toString();
  }
}

File: code/shoring/compat/src/main/kotlin/com/panopset/compat/JavaVersionChart.kt
package com.panopset.compat

import com.panopset.compat.Stringop.getEol
import java.io.StringWriter

class JavaVersionChart {
    private fun print(): String {
        val sw = StringWriter()
        for (mv in MajorVersion.entries) {
            sw.append(getEol())
            sw.append(mv.toString())
        }
        return sw.toString()
    }

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            dspmsglg(chart)
        }

        private val chart = JavaVersionChart().print()
    }
}
